src/Microsoft.Health.Utility/ITimed.cs

```diff
@@ -0,0 +1,15 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Utility;
+
+public interface ITimed : IDisposable
```

Comment:
Seems pretty similar to: 

https://github.com/microsoft/healthcare-shared-components/blob/09135f1a5717707d7d6137e8c5dbe35777dcb9b4/src/Microsoft.Health.Core/ActionTimer.cs#L13-L40

Reply:
Thanks for sharing. I am taking a look at it.

Reply:
I preferred the abstracted version since this class depends on the logger, in our case, it could be a telemetry client.